### DISCUSSION TITLE
[Uczymy się architektury] Spotkanie Architektura w Azure - bezpieczeństwo komunikacji sieciowej #azure-solutions-architect

### DISCUSSION THREAD

**Discussion by aryll** [AzureCommunityPL/group#92] [Wydarzenia]
Cześć 😄 

Uczymy się tworzyć architekturę w Azure! Jeśli myślisz, że to też Twoja ścieżka - dołącz do nas! Zapraszamy osoby o każdym poziomie zaawansowania!

Każde spotkanie to nowe architektoniczne zadanie, które pozwala nam poznać dany fragment Azure - omawiamy nasze propozycje i w ten sposób zdobywamy wiedzę!

Najbliższe spotkanie: **22/10/2020 godzina 20:00**
Tematyka: **zabezpieczenie komunikacji pomiedzy on-prem a Azure**

Treść zadania:
Firma chce zabezpieczyć komunikację sieciową pomiędzy zasobami, które są hostowane na Azure oraz zasobami postawionymi na on-premie. Cała infrastruktura powinna być dostępna tylko z sieci firmowej, oraz w ramach podsieci dla maszyn wirtualnych. W chwili obecnej podjęta została decyzja o przeniesieniu całych maszyn wirtualnych wraz z zasobami natomiast zaproponowane rozwiązanie powinno uwzględnić możliwość użycia różnych zasobów Azure w przyszłości.

Wymagania:
- dwie wersje systemu: 1) stosunkowo bezpieczna i tania oraz 2) droższa ale bardziej zabezpieczona
- należy policzyć koszty rozwiązania (bez kosztów maszyn wirtualnych)
- istotna jest analiza wdrożenia innych usług hostowanych na Azure w przyszłości
- nakład pracy potrzebny do wdrożenia obu rozwiązań
- sposób monitorowania / zarządzania rozwiązaniami
- analiza SLA        

Link do spotkania: 
https://teams.microsoft.com/l/meetup-join/19%3ameeting_MGYzOTZjYWYtNDliNi00NzJiLWFkNWEtYjRhMzJhMGFiMzcw%40thread.v2/0?context=%7b%22Tid%22%3a%22ef60a5a4-900c-4789-89cd-0c42f28ef541%22%2c%22Oid%22%3a%22843f36ea-2af5-4be3-a168-6b8fd3acd1b3%22%7d

Kalendarz:
https://evt.mx/nLArx0BI

W razie pytań/wątpliwości pisz do mnie lub @maciejgos

**Comment by aryll** [comment]
Cześć 😄 

Wrzucajcie swoje pomysły rozwiązania zadania do naszego repo:
https://github.com/AzureCommunityPL/group/tree/master/meetings/azure-solutions-architect

**Comment by maciejgos** [comment]
Cześć 🙂
Już dzisiaj jest spotkanie na którym pogadamy sobie o architekturze. Zachęcamy do wrzucenia swoich rozwiązań do naszego repo link powyżej.
Przykład rozwiązania znajdziecie w https://github.com/AzureCommunityPL/group/blob/master/meetings/azure-solutions-architect/azure-solutions-architect-qa-2020-10-22.md

**Comment by aryll** [comment]
Cześć 😄 

Kilka przydatnych linków do spotkania:
**Service enpoints:** https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-service-endpoints-overview
**Private endpoint:** https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-overview
**Private link:** https://docs.microsoft.com/en-us/azure/private-link/private-link-service-overview
**Łączenie sieci on-premises przez VPN Gateway:** https://docs.microsoft.com/en-us/learn/modules/connect-on-premises-network-with-vpn-gateway
**Łączenie sieci on-premises przez ExpressRoute:** https://docs.microsoft.com/en-us/learn/modules/connect-on-premises-network-with-expressroute

**Comment by miqm** [comment]
Mam uwagę do przedstawionej wyceny rozwiązania - napisaliście że rozwiązanie jest droższe o $9. Wydaje mi się, że jest to błędne, gdyż potrzebujemy trzech Private Endpointów, zatem koszt będzie wynosić $27.

**Reply by aryll** [reply]
Tak naprawdę temat jest trochę bardziej skomplikowany, bo dla private endpointów płacimy za:
- endpoint
- dane wchodzące
- dane wychodzące

czyli w sumie bez założeń związanych z ilością danych, które będą tędy wędrować nie da się tego oszacować.
https://azure.microsoft.com/en-us/pricing/details/private-link/

Drugie pytanie to ile faktycznie potrzebujemy Private Endpoint'ów i Private Linków:
`Multiple private endpoints can be created using the same private link resource. For a single network using a common DNS server configuration, the recommended practice is to use a single private endpoint for a given private link resource to avoid duplicate entries or conflicts in DNS resolution.`

I tu jak widać może też być dowolność 😄 
W tym szacowaniu kosztów bardziej nam chodziło o rozwinięcie tematu że stworzenie Service Endpoint nie pociąga za soba kosztów, natomiast Private Link już tak - faktycznie szkoda, że nie omówiliśmy tego na spotkaniu trochę bardziej szczegółowo.

Happy to be wrong 😄

**Reply by miqm** [reply]
No tak, ale pomijając koszt danych, to za endpoint się płaci za godzinę. Więc mając 3 usługi, nawet jeśli nie będzie żadnego ruchu to i tak zapłacimy za czas posiadania. A odpinanie private linków na noc albo weekendy trąci 🧅

**Reply by maciejgos** [reply]
@miqm W ramach zadania mieliśmy zaprojektować architekturę w której potencjalnie mogą zostać użyte inne usługi oprócz VM. Stąd na diagramach mamy CosmosDB, App Service, itd...
Do wyliczenia przyjęliśmy założenie, że ruch będziemy mieli na poziomie 100GB, usługa będzie dostępna przez 730h (czas ok. 1 miesiąca) oraz jest to koszt **jednego private endpoint**. Wychodzi wtedy dokładnie $9.30. Jak będziemy dokładać kolejne usługi koszt będzie rósł. 
>miesięczny koszt rozwiązania opartego o Private Link to ok. $9. Przy założeniu transferu inbound/outbound na poziomie 100GB, oraz wykorzystaniu jednego "link" dla usług!!!

Przy tego typu wyliczeniach jak trzeba podać jakiś koszt biorę raczej górną wartość 😀

**Reply by miqm** [reply]
@maciejgos - rozumie skąd cena, ale na diagramie są 3 usługi w vnet za private linkiem, zatem potrzebujemy 3 private endpointów do nich, za które trzeba zapłacić godzinowo. Tylko o to mi chodzi.
Jak @aryll napisała, sam private link jest w zasadzie za free jako usługa i kosztuje tylko transfer, ale za private endpointy się płaci od czasu.

**Comment by aryll** [comment]
Podrzucam trochę linków, które pokazywaliśmy wczoraj i takich które mogą Wam się przydać 😄 Podrzucajcie też swoje to zbierzemy je na koniec razem i wrzucimy do repo!

**Egzaminy Microsoft:**
Azure certifications: https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RE4wyqh
Wszystkie certyfikaty z linkami: https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RE2PjDI

**Materiały do nauki:**
Microsoft learn: https://docs.microsoft.com/en-us/learn/
Eventy (też takie gdzie można dostać vouchery na egzaminy): https://events.microsoft.com/?timeperiod=next30Days&isSharedInLocalViewMode=true&country=Poland&language=English
Wtorki z security: https://info.microsoft.com/CE-SCRTY-CATALOG-FY21-07Jul-29-WtorkizSecurity-SRDEM32279_CatalogDisplayPage.html
Czwartki z Azure: https://info.microsoft.com/CE-AzureINFRA-CATALOG-FY20-04Apr-28-ThursdayswithAzure-SRDEM21201_CatalogDisplayPage.html
Pluralsight + Azure: https://www.pluralsight.com/partners/microsoft/azure
Kurs AZ-900 Adama (podstawy Azure): https://www.youtube.com/channel/UCdmEIMC3LBil4o0tjaTbj0w 

**Podstawy do tematu, który omawialiśmy wczoraj:**
NSG & Service Endpoint: https://docs.microsoft.com/en-gb/learn/modules/secure-and-isolate-with-nsg-and-service-endpoints/
VPN Gateway + on-premises: https://docs.microsoft.com/en-gb/learn/modules/connect-on-premises-network-with-vpn-gateway/

**Byłoby super gdybyśmy wspólnymi siłami rozwinęli [notatkę po spotkaniu](https://github.com/AzureCommunityPL/group/blob/master/meetings/azure-solutions-architect/azure-solutions-architect-qa-2020-10-22.md) - róbcie fork i wrzucajcie ciekawe linki, przemyślenia oraz opis tematów które omawialiśmy**

**Comment by aryll** [comment]
Nagranie ze spotkania dostępne już jest w [notatce](https://github.com/AzureCommunityPL/group/blob/master/meetings/azure-solutions-architect/azure-solutions-architect-qa-2020-10-22.md) dzięki @eklime 😄
